pmm-app/src/pmm-settings/Settings.service.ts

```diff
@@ -1,10 +1,52 @@
-import { apiRequest } from '../shared/components/helpers/api';
+import { apiRequest } from 'shared/components/helpers/api';
+import { showSuccessNotification } from 'shared/components/helpers';
+import { Messages } from './Settings.messages';
+import { Settings } from './Settings.types';
+
+export type LoadingCallback = (value: boolean) => void;
 
 export const SettingsService = {
-  getSettings() {
-    return apiRequest.post<any, any>('/v1/Settings/Get', {});
+  async getSettings(callback: LoadingCallback) {
+    let response: any = null;
+
+    try {
+      callback(true);
```

Comment:
Does it make sense to call the callback before the request, not after?

Reply:
The "callback" is the LoadingCallback, to set loading to true/false, so it needs to be both, before and after the request. But maybe I should call it "setLoading" then? To be easier to understand